### PR TITLE
Feature/harden types

### PR DESCRIPTION
**Pull request by doubret**
I created typed interfaces to be returned when calling the `getXXXMetricValue` functions instead of returning `{}`.

The `Gauge` case is more tricky, as it's not a class but an interface it's not possible to check if a given metric is a `Gauge`.

In the end i removed the checks that returned `null` in some cases, it should be the responsibility of the message builder to perform these checks and ignore values or not (that should be business logic).

### CODE REVIEW

**Comment by rstiller** [comment]
here is a function that checks if an object is a Gauge: https://github.com/rstiller/inspector-metrics/blob/159b9d1bd267884bdfbd9f475da188a2af7e5e0b/lib/metrics/metric-registry.ts#L178

File: lib/metrics/AmqpMetricReporter.ts [hunk 1]
```diff
@@ -60,6 +60,69 @@ export interface AmqpMetricReporterOptions extends ScheduledMetricReporterOption
...
+ * Interface used when extracting values from a {@link Gauge}.
+ */
+export interface IGaugeValue<T> {
+  value: T;
+}
+
+/**
+ * Interface used when extracting values from a {@link Histogram}.
+ */
+export interface IHistogramValue {
```
[the remark this answers is not in the document]
Comment (doubret) [reply lib/metrics/AmqpMetricReporter.ts]: I didn’t remembrer this interface.
What fields related to it should we add ?

[the remark this answers is not in the document]
Comment (rstiller) [reply lib/metrics/AmqpMetricReporter.ts]:
```javascript
import { Buckets } from "inspector-metrics";
{
    buckets: Buckets,
    counts: Map<number, number>
}
```

[the remark this answers is not in the document]
Comment (doubret) [reply lib/metrics/AmqpMetricReporter.ts]: done

**Comment by doubret** [comment]
> here is a function that checks if an object is a Gauge: https://github.com/rstiller/inspector-metrics/blob/159b9d1bd267884bdfbd9f475da188a2af7e5e0b/lib/metrics/metric-registry.ts#L178

Yes, that's the best we can do.
But the problem is also about propagating the generic type of `Gauge<T>` , hence the cast `metric as Gauge<any>`.

I'm going to add the check but it won't solve the propagation problem (i tried to return a different type when `T` is a number, in the end i always return the same type, it's a more robust aproach anyway).

**Pull request merged**